pages/_app.js

```diff
@@ -9,10 +9,15 @@ export default function App({ Component, pageProps }) {
   });
 
   function handleCreateEmotion(newEmotion) {
-    setEmotions((prevEmotions) => [
-      { id: uuidv4(), ...newEmotion },
-      ...prevEmotions,
-    ]);
+    setEmotions((prevEmotions) => {
+      const updatedEmotions = [
+        { id: uuidv4(), ...newEmotion },
+        ...prevEmotions,
+      ];
+      return updatedEmotions.sort(
+        (a, b) => new Date(b.dateTime) - new Date(a.dateTime)
+      );
+    });
```

Comment:
We could do the same thing with less code 

``` js
  setEmotions(prevEmotions => [
    { id: uuidv4(), ...newEmotion },
    ...prevEmotions
  ].sort((a, b) => new Date(b.dateTime) - new Date(a.dateTime)));
```

Up to you though. It will depend on how readable you find the suggestion.